src/NUnitFramework/framework/Constraints/Comparers/EquatablesComparer.cs

```diff
@@ -48,6 +49,10 @@ internal EquatablesComparer(NUnitEqualityComparer equalityComparer)
             Type xType = x.GetType();
             Type yType = y.GetType();
 
+            if (x is IEnumerable && DoesUseStructuralEquality(xType)
+                && y is IEnumerable && DoesUseStructuralEquality(yType))
+                return null;
```

Comment:
Skip using EquatablesComparer if each type fulfills these criteria:
1. Implements IEnumerable
2. Implements either IStructuralComparable or IStructuralEquatable
3. Is a value type

I'm unsure if this is overkill. Would it be enough to just check for `IEnumerable` here, or would that be overly broad and break something? Alternately, would it be better to instead check if the types are valuetypes that implement `IStructuralEquatable` (though I suspect this might break the "defer to enumerables" idea)?

Reply:
I like the idea that we would give IStructural interfaces priority over IEnumerable. Not sure that being a value type should make a difference; is there are reason you can think of?

Reply:
Good points thanks @jnm2 . I think I'd been looking for a "quick and easy" fix here, but I see it's a little more nuanced. Especially with having IStructural take priority over IEnumerable (similar to how EquatablesComparer takes priority).

Re: value types, I had originally added that in error. I noticed MSDN primarily showed value types as implementing this, but of course both structs and classes can implement interfaces.

I've stepped back and given this a different approach now, with a new `StructuralComparer` type which, when invoked, will use NUnitEqualityComparer for all comparisons (including Tolerance).